### DISCUSSION TITLE
uniapp - 好记性不如烂笔头

### DISCUSSION THREAD

**Discussion by giscus** [xiaodongxier/wiki#10] [Announcements]
# uniapp - 好记性不如烂笔头

王永杰的个人网站,王永杰的个人wiki,小东西儿的个人网站,xiaodongxier.com,wangyongjie.cn,前端知识管理,个人wiki,个人知识管理

https://wiki.wangyongjie.cn/%E7%A7%BB%E5%8A%A8%E7%AB%AF/uniapp/

**Comment by xiaodongxier** [comment]
uni-app/小程序绑定style对象渲染结果是[object Object]？
https://blog.csdn.net/flow_camphor/article/details/120042927
https://developers.weixin.qq.com/community/develop/doc/0006a87b8089f87514ce62e5856400

**Comment by xiaodongxier** [comment]
目前的项目没有uni_modules文件夹是不是就不是uni_modules版本？如果非uni_modules版本转uni_modules版本，需要做什么？对目前的项目有什么影响？

**Comment by xiaodongxier** [comment]
微信小程序拉开输入框与键盘的距离:https://blog.csdn.net/weixin_43755513/article/details/105220389
```
<input cursor-spacing="100"></input>
```

**Reply by xiaodongxier** [reply]
https://blog.csdn.net/zjy_android_blog/article/details/102668261

**Reply by xiaodongxier** [reply]
https://blog.csdn.net/xyphf/article/details/134073110

**Reply by xiaodongxier** [reply]
padding-bottom: constant(safe-area-inset-bottom); /*兼容 IOS<11.2*/
			padding-bottom: env(safe-area-inset-bottom); /*兼容 IOS>11.2*/

**Reply by xiaodongxier** [reply]
重点部分
```
		<view class="chat-input-view">
			<view class="input-select">
				<view class="select-item" @click="handleClickNewChat">开启新对话</view>
			</view>
			<view class="input-item" :style="{ paddingBottom: bottomPadding + 'px' }">
				<input
					cursor-spacing="10"
					v-model="inputValue"
					:type="type"
					:adjust-position="true"
					placeholder="请输入内容"
					@confirm="sendClick"
				/>
				<view class="send-btn">
					<button
						class="mini-btn"
						type=""
						size="mini"
						@touchend.prevent="sendClick"
					>
						发送
					</button>
				</view>
			</view>
		</view>
```

```
data(){
isAndroid: false, // 是否是安卓平台
			bottomPadding: 30 // 默认底部间距
}
```


```
	onLoad(options) {
		// 适配小米等安卓设备
		const systemInfo = this.$u.sys(); // 使用 uView 提供的 sys() 获取系统信息
		this.isAndroid = systemInfo.platform === "android";
		if (this.isAndroid) {
			this.bottomPadding = 20; // 如果是 Android，手动设置底部间距
		} else {
			this.bottomPadding = systemInfo.safeAreaInsets.bottom || 30; // iOS 根据安全区域适配
		}
	},
```
